### PR TITLE
Add asset download endpoint, and change asset path filter mode

### LINKED ISSUE #52: Add asset download endpoint
This should be consumed by #37.

### PR DESCRIPTION
**Pull request by AlmightyYakob**
Fixes #52.

The reason for changing the lookup mode of the `path_prefix` AssetFilter field (now just `path`), is because:
* It wasn't being used for prefix searching
* It's needed for use by the frontend when fetching the download link for a given asset path (i.e. turning a path to an asset into the actual asset information).

### CODE REVIEW

File: publish/views/asset.py [hunk 1]
```diff
@@ -56,6 +57,11 @@ class AssetViewSet(NestedViewSetMixin, DetailSerializerMixin, ReadOnlyModelViewS
     filter_backends = [filters.DjangoFilterBackend]
     filterset_class = AssetFilter
 
+    @action(detail=True, methods=['GET'])
+    def download(self, request, **kwargs):
+        """Return a redirect to the file download in S3."""
```
Comment (mgrauer) [inline publish/views/asset.py:62]: I'd say to remove S3 as it is an implementation detail and may not make sense for e.g. local deployments.

`"""Return a redirect to the file download in the object store."""`

Comment (AlmightyYakob) [reply publish/views/asset.py:62]: Good catch

Comment (AlmightyYakob) [reply publish/views/asset.py]: ea1227f

**Comment by brianhelba** [comment]
@AlmightyYakob 
> It's needed for use by the frontend when fetching the download link for a given asset path (i.e. turning a path to an asset into the actual asset information).

Is it possible to just use the UUID internally in the frontend? This would be much more robust.

File: publish/views/asset.py [hunk 2]
```diff
@@ -34,11 +35,11 @@ class Meta(AssetSerializer.Meta):
 
 
 class AssetFilter(filters.FilterSet):
-    path_prefix = filters.CharFilter(lookup_expr='startswith', field_name='path')
+    path = filters.CharFilter(lookup_expr='exact', field_name='path')
```
Comment (brianhelba) [inline publish/views/asset.py:38]: I don't think you need to explicitly define a `path =` filter at all. The fact that we've requested `fields = ['path']` *should* be enough for it to auto-generate a standard filter from the model.

**Comment by AlmightyYakob** [comment]
> @AlmightyYakob
> 
> > It's needed for use by the frontend when fetching the download link for a given asset path (i.e. turning a path to an asset into the actual asset information).
> 
> Is it possible to just use the UUID internally in the frontend? This would be much more robust.

This is what's currently done once the asset information has been fetched. However, the `path` filter is used to turn a path from the `paths` endpoint into an asset object, which contains the assets `UUID`. I had initially thought about returning whole asset objects from the `paths` endpoint, to allow for easier retrieval of this information, but I decided against it because that endpoint doesn't return assets themselves, it returns partial paths, and I didn't want to arbitrarily mix two different data models in that endpoint's return.

File: publish/views/asset.py (hunk 2 above)
Comment (AlmightyYakob) [reply publish/views/asset.py:38]: Good point, I'll change it and make sure it behaves like we expect.

Comment (AlmightyYakob) [reply publish/views/asset.py]: e9a1aaf

File: publish/views/asset.py [hunk 3]
```diff
@@ -34,11 +35,9 @@ class Meta(AssetSerializer.Meta):
 
 
 class AssetFilter(filters.FilterSet):
-    path_prefix = filters.CharFilter(lookup_expr='startswith', field_name='path')
-
     class Meta:
         model = Asset
-        fields = ['path_prefix']
```
Comment (brianhelba) [inline publish/views/asset.py:41]: @AlmightyYakob One final question: the removal of `path_prefix` as a general filter doesn't affect the fact that `path_prefix` is used as a query string within the `paths` action?

Comment (AlmightyYakob) [reply publish/views/asset.py:41]: No, it's just a coincidence, and part of the reason why I wanted to replace the `path_prefix` filter.

**Pull request merged**